src/service/variable-source.js

```diff
@@ -41,7 +42,16 @@ let ReplacementDef;
  * @return {!Promise<ResolverReturnDef>}
  */
 export function getTimingDataAsync(win, startEvent, endEvent) {
-  return loadPromise(win).then(() => {
+  const deferred = new Deferred();
+  loadPromise(win).then(() => {
```

Comment:
would the following work?
```js
loadPromise(win)
    .then(timer.promise(1))
    .then(...);
```

Reply:
That's should work, as long as there's a `setTimeout` call involved. But do you think it's worth retrieving the `Timer` service here?

Reply:
Yep, the whole point of timer service is to share code.

Reply:
BTW: I found a `delayAfterDeferringToEventLoop` method from the lightbox-gallery component and few use case of `timer.promise()`. May worth introducing a global helper method for this. But I'll stick to timer.promise here.

Reply:
done